pulsar-client-api/src/main/java/org/apache/pulsar/client/api/ProducerBuilder.java

```diff
@@ -434,4 +434,18 @@
      * @return the producer builder instance
      */
     ProducerBuilder<T> autoUpdatePartitions(boolean autoUpdate);
+
+    /**
+     * Option to control whether enable multiple schema mode,
+     * which would be enabled automatically if this feature is touched by default.
+     * If enabled, producer can send a message with different schema from that specified just when it is created,
+     * otherwise a invalid message exception would be threw if the producer want to send a message
+     * with different schema.
+     *
+     * @param switchOption
+     *            indicates to enable, disable, or automatically enable multiple schema mode
+     * @return the producer builder instance
+     * @since 2.5.0
+     */
+    ProducerBuilder<T> withMultiSchema(SwitchOption switchOption);
```

Comment:
`SwitchOption` is an unclear name. I would suggest using something more specific. for example, 'MultiSchemaProduceMode'.

Reply:
Great, I meant a generic name. It's good, I'll fix it.

Reply:
fixed, PTAL

Reply:
Outdated for signature changed.